server/src/main/java/de/uftos/builders/SpecificationBuilder.java

```diff
@@ -0,0 +1,41 @@
+package de.uftos.builders;
+
+import java.util.List;
+import java.util.Optional;
+import org.springframework.data.jpa.domain.Specification;
+
+public class SpecificationBuilder<T> {
+
+  private Specification<T> specification;
+
+  public SpecificationBuilder() {
+    specification = Specification.where(null);
+  }
+
+  public Specification<T> build() {
+    return specification;
+  }
+
+  public SpecificationBuilder<T> addOptionalOr(Optional<String> param, String paramName) {
```

Comment:
IMO the `add` can be omitted here, so just `optionalOr` (same for the other methods)

Reply:
Now that I think of it, I like your proprosition better